### PR TITLE
drivers: ad7606: misc changes to provide support for IIO-example project

### PR DESCRIPTION
**Pull request by commodo**
## Pull Request Description

Misc changes added before adding the IIO example project.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Review by dbogdan — DISMISSED** [review]

**Review by amiclaus — APPROVED** [review]

File: drivers/adc/ad7606/ad7606.h [hunk 1]
```diff
@@ -232,8 +243,8 @@ struct ad7606_range {
 	int32_t min;
 	/** Maximum range value */
 	int32_t max;
-	/** Whether the range is differential */
-	bool differential;
+	/** Type of range according to @enum ad7606_range_type */
```
Comment (amiclaus) [inline drivers/adc/ad7606/ad7606.h:246]: seems like doxygen is complaining about how you reference here the range_type enum. Please use the doxygen commands for this:
https://www.doxygen.nl/manual/commands.html#cmdref

File: drivers/adc/ad7606/ad7606.c [hunk 2]
```diff
@@ -1495,8 +1495,11 @@ int32_t ad7606_set_config(struct ad7606_dev *dev,
 	if (dev->sw_mode) {
 
 		val |= no_os_field_prep(AD7606_CONFIG_OPERATION_MODE_MSK, config.op_mode);
-		if ((uint8_t)config.dout_format > (uint8_t)dev->max_dout_lines)
+		if ((uint8_t)config.dout_format > (uint8_t)dev->max_dout_lines) {
+			printf("Device misconfig: max DOUT is %u, but trying to set %u\n",
```
Comment (buha) [inline drivers/adc/ad7606/ad7606.c:1538]: If the project that uses this device driver has an running iiod that communicates with an iio client over serial, the comm will be messed up by these messages. It's pretty common that we redirect printf output to serial. I'd drop this one change.

**Comment by buha** [comment]
the rest looks good

**Comment by commodo** [comment]
Changelog v2 -> v3:
* rebased on newer main

**Comment by commodo** [comment]
Changelog v3 -> v4:
* dropped commit `drivers: ad7606: print a message when DOUT lines misconfigured`

**Review by buha — APPROVED** [review]

**Pull request merged**